### ISSUE TITLE
npm view alternative

### ISSUE THREAD

**Issue by juanpicado** [tavvy/verdaccio-badges#2]
Great stuff :) 👏 

I've notice you are spawn a process in order to get the latest version. I was wondering if you have considered add the `--json` flag like:

```
npm view lodash --registry https://registry.npmjs.org --update-notifier=false --json
```

You can actually does a fetch call and get rid of the `spawn` if you like. I wonder your thoughts about this.

**Comment by tavvy** [comment]
@juanpicado firstly - really appreciate the input 👍 

I did try with the `--json` flag, but it required an additional `JSON.parse()` on the result which seemed like overhead, although would have the benefit of being more defensive

I did also consider a fetch but liked the idea of keeping it native and not pulling in any extra deps than required. Although this would have the benefit of not requiring additional config / ownership changes on the `~/.npm` folder.

So truth to be honest im still undecided on it.
If I share the same http req module as the root project then I can likely go down the fetch route. Looking at https://github.com/verdaccio/verdaccio/blob/master/package.json#L22 the request library you use isn't immediately obvious, is it `request`?

**Comment by juanpicado** [comment]
Yes, we are using `request`, but since is deprecated I'd not recommend use it. I also plan to change it in next major release. But we are heavily using it in multiple places https://github.com/verdaccio/monorepo/blob/master/plugins/audit/src/audit.ts#L32 

My main point is, if npm changes the presentation text in any way your plugin will be broken if they decided in npm7 to change that, still you will need to provide backward support for older npm, so for me if you want to keep it native,  the `JSON.parse()` would be the less worse. I also like to have less dependencies as possible.

**Comment by juanpicado** [comment]
Furthermore, if you do the request directly to the registry instead use `npm view` which internally also calls the same endpoint, parse and make slow your request anyway.

**Comment by tavvy** [comment]
Cool, thanks for your suggestions 👍
